File: NextUse.Solution/NextUse.DAL/Database/Entities/Address.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NextUse.DAL.Database.Entities
{
    public class Address
    {
        public int Id { get; set; }
        public required string Country { get; set; }
        public required string City { get; set; }
        public int PostalCode { get; set; }
        public string? Street { get; set; }
        public string? HouseNumber { get; set; }

        public Profile? Profile { get; set; }
        public Product? Product { get; set; }


    }
}

File: NextUse.Client/src/app/_models/address.ts
import { Profile } from "./profile";
import { Product } from "./product";

export interface Address {
    id: number;
    country: string;
    city: string;
    postalCode: number;
    street?: string;
    houseNumber?: string;

    profileId?: number;
    profile?: Profile;
    productId?: number;
    product?: Product;
}